python/GangaDirac/Lib/Backends/DiracBase.py

```diff
@@ -536,45 +543,57 @@ def _internal_job_finalisation(job, updated_dirac_status):
         logger = getLogger()
 
         if updated_dirac_status == 'completed':
+            import time
+            start = time.time()
             # firstly update job to completing
             DiracBase._getStateTime(job, 'completing')
             if job.status in ['removed', 'killed']:
                 return
-            if (job.master and job.master.status in ['removed', 'killed']):
+            elif (job.master and job.master.status in ['removed', 'killed']):
                 return  # user changed it under us
 
             job.updateStatus('completing')
             if job.master:
                 job.master.updateMasterJobStatus()
 
-            import time
-            start = time.time()
             # contact dirac for information
-            job.backend.normCPUTime = execute('normCPUTime(%d)' % job.backend.id)
-            getSandboxResult = execute("getOutputSandbox(%d,'%s')" % (job.backend.id, job.getOutputWorkspace().getPath()))
-            file_info_dict = execute('getOutputDataInfo(%d)' % job.backend.id)
+            #job.backend.normCPUTime = execute('normCPUTime(%d)' % job.backend.id)
+            #getSandboxResult = execute("getOutputSandbox(%d,'%s')" % (job.backend.id, job.getOutputWorkspace().getPath()))
+            #file_info_dict = execute('getOutputDataInfo(%d)' % job.backend.id)
+
+            output_path = job.getOutputWorkspace().getPath()
+
+            job.backend.normCPUTime, getSandboxResult, file_info_dict = execute("finalize_job(%d, '%s')" % (job.backend.id, output_path))
+
             now = time.time()
             logger.debug('Job ' + job.fqid + ' Time for Dirac metadata : ' + str(now - start))
 
-            logger.debug('Job ' + job.fqid + ' OutputDataInfo: ' + str(file_info_dict))
-            logger.debug('Job ' + job.fqid + ' OutputSandbox: ' + str(getSandboxResult))
+            #logger.info('Job ' + job.fqid + ' OutputDataInfo: ' + str(file_info_dict))
+            #logger.info('Job ' + job.fqid + ' OutputSandbox: ' + str(getSandboxResult))
+            #logger.info('Job ' + job.fqid + ' normCPUTime: ' + str(job.backend.normCPUTime))
 
             # Set DiracFile metadata
             wildcards = [f.namePattern for f in job.outputfiles.get(DiracFile) if regex.search(f.namePattern) is not None]
 
-            with open(os.path.join(job.getOutputWorkspace().getPath(), getConfig('Output')['PostProcessLocationsFileName']), 'ab') as postprocesslocationsfile:
+            lfn_store = os.path.join(output_path, getConfig('Output')['PostProcessLocationsFileName'])
+
+            if not os.path.isfile(lfn_store):
+                with open(lfn_store, 'w'):
+                    pass
```

Comment:
Yes, although this is used in enough other places in Ganga I assumed people were familiar with it.